### PR TITLE
Update otelopscol

### PR DESCRIPTION
**Pull request by braydonk**
## Description
This PR updates the version of the `opentelemetry-operations-collector` submodule for the purpose of picking up the update to otel v0.102.0 dependencies.

## Related issue
[b/339060391](http://b/339060391)

## How has this been tested?
Integration tests in this PR.

## Checklist:
- Unit tests
  - [x] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [x] Integration tests do not apply.
  - [ ] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

**Review by jefferbrecht — APPROVED** [review]

File: dockerfiles/template
Comment (jefferbrecht) [inline dockerfiles/template:83]:
```suggestion
{package_build} || true &> /dev/null
```

Comment (ridwanmsharif) [reply dockerfiles/template]: Done.

**Comment by ridwanmsharif** [comment]
I temporarily cherrypicked https://github.com/GoogleCloudPlatform/ops-agent/pull/1736 on top of this PR to see if the tests pass.

Tomcat and Jetty tests failed and I filed b/347107292. I'm not sure if its related to the OTel upgrade or if its also failing on HEAD (I say this cause its only failing on the new ML image). My suspicion is the latter but I'll look in the morning.

I'm also dismissing the old review since there were 1456782 commits since this was reviewed last

**Comment by ridwanmsharif** [comment]
The nvml tests failed because of  quota issues. The Windows integration test seems like a flake (passed on earlier runs and was an error creating the VM).

@jefferbrecht mind giving another review before we merge? If you're busy, I'll assign it to a random reviewer

**Review by jefferbrecht — APPROVED** [review]

**Comment by ridwanmsharif** [comment]
Test failures are unrelated, but we should really look into b/347107292. I consistently happens now on master too

**Pull request merged**
